examples/grid_sequencer/sketch.js

```diff
@@ -0,0 +1,162 @@
+  for (var i = 0; i < numTimeSteps; i++) {
+    for (var j = 0; j < pitches.length; j++) {
+      var x = i * cellWidth;
+      var y = controlPanelHeight + j * cellHeight;
+      var pitch = pitches[pitches.length - j - 1]; // Pitches go from bottom to top
+      var newCell = new Cell(createVector(x, y), pitch);
+      cells.push(newCell);
+    }
+  }
+
+  // Create a synth to make sound with
+  synth = new p5.PolySynth();
+
+  // Create SoundLoop with 8th-note-long loop interval
+  sloop = new p5.SoundLoop(soundLoop, "8n");
+  sloop.bpm = bpm;
+
+  // UI
+  playPauseButton = createButton('PLAY/PAUSE');
+  playPauseButton.mousePressed(togglePlayPause);
+  playPauseButton.position(0, 0);
+  playPauseButton.size(width / 4, controlPanelHeight);
+
+  tempoSlider = createSlider(30, 300, bpm);
+  tempoSlider.position(width / 4, 0);
+  tempoSlider.size(width / 4, controlPanelHeight);
+  tempoText = createP("BPM: " + bpm);
+  tempoText.position(width / 2, 0);
+  tempoText.size(width / 4, controlPanelHeight);
+
+  clearButton = createButton('CLEAR ALL');
+  clearButton.mousePressed(clearAll);
+  clearButton.position(width * 3 / 4, 0);
+  clearButton.size(width / 4, controlPanelHeight);
+}
+
+function soundLoop(cycleStartTime) {
+  for (var i = 0; i < cells.length; i++) {
+    if (floor(i / pitches.length) == timeStepCounter) {
+      cells[i].active = true;
+      if (cells[i].enabled) {
+        // Play sound
+        var velocity = 1; // Between 0-1
+        var quaverSeconds = this._convertNotation('8n'); // 8th note = quaver duration
+        var freq = midiToFreq(cells[i].pitch);
+        synth.play(freq, velocity, cycleStartTime, quaverSeconds);
+      }
+    } else {
+      cells[i].active = false;
+    }
+  }
+  this.bpm = bpm;
+  timeStepCounter = (timeStepCounter + 1) % numTimeSteps;
+}
+
+function draw() {
+  background(255);
+  for (var i = 0; i < cells.length; i++) {
+    cells[i].checkIfHovered();
+    cells[i].display();
+  }
+  bpm = tempoSlider.value();
+  tempoText.html("BPM: " + bpm);
+}
+
+function mouseClicked() {
+  for (var i = 0; i < cells.length; i++) {
+    if (cells[i].hovered) {
+      cells[i].enabled = !cells[i].enabled;
+    }
+  }
+}
+
+function togglePlayPause() {
+  if (sloop.isPlaying) {
+    sloop.pause();
+  } else {
+    sloop.start();
+  }
+}
+
+function clearAll() {
+  for (var i = 0; i < cells.length; i++) {
+    cells[i].enabled = false;
+  }
+}
+
+
+var Cell = function (position, pitch) {
+  // Sound
+  this.pitch = pitch;
+  // Appearance
+  this.padding = 2;
+  this.position = position.copy();
+  this.width = cellWidth - 2 * this.padding;
+  this.height = cellHeight - 2 * this.padding;
+  this.defaultColor = [190, 240, 255];
+  // Mouse hover
+  this.hovered = false;
+  this.hoverColor = [230, 255, 255];
+  // Enabled when clicked
+  this.enabled = false;
+  var varyingColorVal = 22 * (this.pitch % pitches.length);
+  this.enabledColor = [20 + varyingColorVal, 255 - varyingColorVal, 255];
```

Comment:
One of the `enabledColor` is hard for me to distinguish from the `defaultColor`. It should be able to work in greyscale—there are chrome extensions like [this](https://www.funkify.org) to test this kind of thing.

`colorMode(HSB)` might make this color scheme easier, where "disabled" always has a different brightness than "enabled"